phabfive/maniphest.py

```diff
@@ -81,10 +96,149 @@ def create_from_config(self, config_file, dry_run=False):
+                has_errors = True
+
+            if has_errors:
+                log.error(f"Could not find one or more specified users, projects in phabricator instance. All available project names")
+                log.error(list(user_name_to_id_mapping.keys()))
+                log.error(f"All available users names")
+                log.error(list(user_name_to_id_mapping.keys()))
+                return (False, None)
+
+        for index, ticket_data in enumerate(data["tickets"]):
+            if ("subscribers" not in ticket_data):
+                break
+
+            if not isinstance(ticket_data["subscribers"], list):
+                log.error(f"data key 'subscribers' must be of list type")
+                return (False, None)
+
+            has_errors = False
+            
+            for subscriber in ticket_data["subscribers"]:
+
+                if subscriber not in user_name_to_id_mapping:
+                    log.error(f"Subscriber user name '{subscriber}' not found in Phabricator instance")
+                    has_errors = True
+
+            if has_errors:
+                log.error(f"Could not find one or more specified users, projects in phabricator instance. All available project names")
+                log.error(list(project_name_to_id_mapping.keys()))
+                log.error(f"All available users names")
+                log.error(list(user_name_to_id_mapping.keys()))
+                return (False, None)
+
+        ### Validate priority is consistent with Phabricator instance, default = TICKET_PRIORITY_NORMAL
+        for index, ticket_data in enumerate(data["tickets"]):
+            if ("priority" not in ticket_data):
+                break
+            if ticket_data["priority"] not in TICKET_PRIORITIES:
+                log.error(f"Priority '{ticket_data['priority']}' not found in Phabricator instance")
+                has_errors = True
+
+            if has_errors:
+                log.error(f"Could not find one or more specified users, projects in phabricator instance. All available project names")
+                log.error(list(project_name_to_id_mapping.keys()))
+                log.error(f"All available users names")
+                log.error(list(user_name_to_id_mapping.keys()))
+                return (False, None)
+        
+        task_id_to_phid_mapping = {}
+
+        ###FIXME: Add parent task
+        for index, ticket_data in enumerate(data["tickets"]):
+            if ("parents" not in ticket_data):
+                break
+
+            if not isinstance(ticket_data["parents"], list):
+                log.error(f"data key 'parents' must be of list type")
+                return (False, None)
+
+            has_errors = False
+
+            constraints = {}
+            constraints["ids"] = []
+            
+            for parent in ticket_data["parents"]:
+                if not re.match(r'^(T).*\d$$', parent):
```

Comment:
This regex looks like some standard regex that might be better suited in constants.py as a stable pattern to look for, i dont really get either what it looks for here so a comment line might be good

Reply:
on reviewing how it is done by you on others e.g. paste and diffusion, instead of re-inventing the wheel I created 

 ```
   def _validate_identifier(self, repo_id):
        return re.match(f"^{MONOGRAMS['maniphest']}$", repo_id)
```

to handle validation and also subsequent removal of T from the whole monogram and converting it into int.

since we are on this, please consider how would you want to `maniphest_app.info(task_id)` - should all be made uniform - passing monogram/str i.e. like how `maniphest_app.add_comment(ticket_identifier)` is implemented with proper validation within the function call  `maniphest_app.info()` instead of stripping the T before passing into  `maniphest_app.info(task_id)` like now in cli.py?

Reply:
Yeah we have thought something similar our self. What we might go with instead tho is to make internal models for some of these primitives like a ticket and put all logic and validation within them instead so that we can pass around proper maniphest objects and re-use logic and features from other tools like Django that we would take inspiration from with how they can lazy load objects and data from a database or api backend like here. Passing around strings this way is not optimal one bit and it needs to be fully refactored eventually

Reply:
in that case... I will do minimal changes for the time being ;-D